### ISSUE TITLE
hlint and `record`

### ISSUE THREAD

**Issue by magthe** [ndmitchell/hlint#133]
`hlint` seems to have problems with the following kind of type definition, which uses [record](http://hackage.haskell.org/package/record):

```haskell
{-# LANGUAGE QuasiQuotes #-}
{-# LANGUAGE DataKinds #-}

import Record

type T = [r| { foo :: Int, bar :: Bool } |]
```

The result of `hlint -v test.hs` is:

```
test.hs:3:12: Warning: Parse error: |
Found:
    import Record
    
  > type T = [r| { foo :: Int, bar :: Bool } |]

1 suggestion
```

**Comment by ndmitchell** [comment]
This is a bug in Haskell-src-exts. The good news is that there's a patch upstream. The bad news is that it hasn't been merged yet. https://github.com/haskell-suite/haskell-src-exts/pull/208

**Comment by magthe** [comment]
We'll just wait until upstream fixes the issue then :)

**Comment by ndmitchell** [comment]
Hmm, I tried the example and it's still broken. Since I can't reopen the last ticket, I've raised a new HSE ticket at https://github.com/haskell-suite/haskell-src-exts/issues/289.

**Comment by mpickering** [comment]
With which version of HSE?

**Comment by ndmitchell** [comment]
@mpickering 1.17.0. I can upgrade if you think 1.17.1 will fix it.

**Comment by mpickering** [comment]
The changes haven't been released yet but they are on the master branch.

**Comment by ndmitchell** [comment]
Ah, then I look forward to the next release :)

**Comment by Blaisorblade** [comment]
Happened onto this thread and noticed there's been three HSE releases since then, so maybe this is fixed?

**Comment by ndmitchell** [comment]
@Blaisorblade - thanks for spotting - it is now fixed.
